core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -251,6 +251,14 @@ private[spark] object JettyUtils extends Logging {
   private def attachPrefix(basePath: String, relativePath: String): String = {
     if (basePath == "") relativePath else (basePath + relativePath).stripSuffix("/")
   }
+
+  /**
+   * Attach a prefix to the given path, but avoid returning an empty path.
+   * Don't remove the trailing / if present.
+   */
+  private def attachPrefixForRedirect(basePath: String, relativePath: String): String = {
+    if (basePath == "") relativePath else (basePath + relativePath)
```

Comment:
I know you're copying & pasting the code from above, but I wonder how is this any different than just always returning `basePath + relativePath`.